solver/jobs.go

```diff
@@ -280,6 +283,39 @@ func NewSolver(opts SolverOpt) *Solver {
 	return jl
 }
 
+func (jl *Solver) findMergeDirection(dest *edge, src *edge) (*edge, *edge) {
```

Comment:
Would it be clearer if instead we used langauge "find if any sibling edge has already owned by the vertex of the targetedge"

Reply:
Have refactored this into a `hasOwner` helper, then the caller determines whether to switch.